youtube_dl/extractor/switchtube.py

```diff
@@ -0,0 +1,181 @@
+                break
+
+        creator_groups = self._search_regex(
+            r'''(?s)<span\b[^>]+?\bclass\s*=\s*("|')(?:(?!\1).)*\bp-author\b(?:(?!\1).)*\1\s*property\s*=\s*["']dc:creator\b[^>]+>\s*(.*?<span\b[^>]+?\bclass\s*=\s*["']p-name\b.*</span>).*?</span>''',
+            webpage, 'creator groups', default='', group=2)
+        creator_groups = re.match(r'''(?s)<a\b[^>]+?\bhref\s*=\s*("|')/profiles/(?P<profile_id>.+?)\1[^>]*>\s*<span\b[^>]+>\s*(?P<creator_name>.+?)\s*</span>[,\s]*<span\b[^>]+\bclass\s*=\s*["']p-organization-name\b[^>]+>\s*(?P<organization_name>.+?)\s*</span>''', creator_groups)
+        if creator_groups:
+            creator_groups = creator_groups.groupdict()
+            for k, v in creator_groups.items():
+                if not v:
+                    # use a default
+                    creator_groups[k] = '[unknown]'
+            info['uploader'] = info['creator'] = ' from '.join((creator_groups['creator_name'], creator_groups['organization_name']))
+            info['uploader_id'] = creator_groups['profile_id']
+            info['uploader_url'] = 'https://tube.switch.ch/profiles/' + info['uploader_id']
+
+        parsed_media_entries = self._parse_html5_media_entries(url, webpage, video_id)[0]
+        info['thumbnail'] = parsed_media_entries['thumbnail']
+        info['subtitles'] = parsed_media_entries['subtitles']
+        info['formats'] = parsed_media_entries['formats']
+        self._sort_formats(info['formats'])
+
+        return info
+
+
+class SwitchTubeChannelOrProfileIE(InfoExtractor):
+    _VALID_URL = r'https?://tube\.switch\.ch/(channels|profiles)/(?P<id>[0-9a-zA-Z]+)'
+    IE_NAME = 'switchtube:channel_or_profile'
+    _TESTS = [{
+        'url': 'https://tube.switch.ch/channels/1deb03e0',
+        'info_dict': {
+            'id': '1deb03e0',
+            'title': 'CS-456 Artificial Neural Networks',
+            'description': 'Class on Artificial Neural Networks and Reinforcement Learning designed for EPFL master students in CS and related disciplines.'
+        },
+        'playlist_mincount': 94,
+    }, {
+        'url': 'https://tube.switch.ch/profiles/94248',
+        'info_dict': {
+            'id': '94248',
+            'title': 'Wulfram Gerstner',
+        },
+        'playlist_mincount': 94,
+    }]
+
+    @classmethod
+    def suitable(cls, url):
+        return False if SwitchTubeIE.suitable(url) else super(SwitchTubeChannelOrProfileIE, cls).suitable(url)
+
+    def _real_extract(self, url):
+        channel_id = self._match_id(url)
+        webpage = self._download_webpage(url, channel_id)
+        channel_title = self._html_search_regex(r'(?s)<title\b[^>]*>(.+?)</title>', webpage, 'title', default=None)
+
+        description = self._html_search_regex(r'''<div\b[^>]+class=("|')p-summary formatted\1[^>]+property=("|')dc:description\2>\s+(.+?)\s+</div>''',
+                                              webpage, 'description', flags=re.DOTALL | re.MULTILINE, default=None, group=3)
```

Comment:
Do you really want `re.MULTILINE`? That means that it matches each line as a separate string. Also, the embedded version is better than flags, and the whole RE can be relaxed:
```suggestion
        description = self._html_search_regex(r'''(?s)<div\b[^>]+class\s*=\s*("|')p-summary formatted\1[^>]+property\s*=\s*("|')dc:description\2[^>]*>\s*(.+?)\s*</div>''',
                                              webpage, 'description', default=None, group=3)
```
If what we want is to get a `None` description when its field is white-space, this fragment `>\s*(.+?)\s*<` doesn't quite work when there are n=2 (3 with `\s+`) or more white-space characters. The RE part inside `>...<` will match the first n-1 characters, the nth and nothing, respectively; the method will strip the matched group and return `''`. So in that case:
```suggestion
        description = self._html_search_regex(r'''(?s)<div\b[^>]+class\s*=\s*("|')p-summary formatted\1[^>]+property\s*=\s*("|')dc:description\2[^>]*>\s*(.+?)\s*</div>''',
                                              webpage, 'description', default=None, group=3) or None
```
Or:
```suggestion
        description = self._html_search_regex(r'''(?s)<div\b[^>]+class\s*=\s*("|')p-summary formatted\1[^>]+property\s*=\s*("|')dc:description\2[^>]*>\s*(\S.*?)\s*</div>''',
                                              webpage, 'description', default=None, group=3)
```